modules/scrollback/index.js

```diff
@@ -466,22 +478,26 @@ module.exports.getFormattedScrollbackLinesFromRanges = function(channel, input,
         for(var c=0; c < nickChunks.length; c++) {
           var chunk = nickChunks[c];
 
-          var matches = chunk.filter(function(el) {
+          var matches = chunk.contents.filter(function(el) {
             return regex.test(module.exports.formatLine(el));
           });
 
           if(matches.length == 0) {
-            //TODO: Guess which nick we got context for
-            return cb("Cannot find line matching regex " + regex.toString() + "; only have " + chunk.length + " of context");
+            error = "Cannot find line";
+            if(chunk.nick !== undefined) {
+              error += " from " + chunk.nick;
+            }
+            error += " matching regex " + regex.toString() + "; only have " + revCache.length + " lines of context";
+            return cb(error, null);
           }
           else {
-            regexChunks.push(matches);
+            regexChunks.push({nick:chunk.nick, regex:regex, contents:matches});
           }
         }
       }
 
       if(regexChunks.length == 0) {
-        regexChunks.push(revCache);
+        regexChunks = nickChunks;
```

Comment:
so if no lines match the nickChunks, is it more correct to return all the nickChunks or  return []?